File: test_gemini.py
import google.generativeai as genai
import os
import sys

# Get API key from env or argument
api_key = os.environ.get("GEMINI_API_KEY")
if not api_key and len(sys.argv) > 1:
    api_key = sys.argv[1]

if not api_key:
    # Fallback to the known key if not found in env
    api_key = "AIzaSyCfDLK5EvsJaIkBf-CcTRA05QnZVa-K54o"

print(f"Testing Gemini API with key: {api_key[:10]}...")

try:
    genai.configure(api_key=api_key)
    
    # List models
    print("\nAvailable models compatible with gemini-1.5-flash:")
    for m in genai.list_models():
        if 'generateContent' in m.supported_generation_methods:
            print(f"- {m.name}")
    
    # Test generation
    print("\nTesting generation with 'gemini-1.5-flash'...")
    model = genai.GenerativeModel('gemini-1.5-flash')
    response = model.generate_content("Hello, can you hear me?")
    print(f"\nResponse: {response.text}")
    print("\n✅ API Key and Model are working!")
    
except Exception as e:
    print(f"\n❌ Error: {e}")

File: backend/test-gemini.js
import { GoogleGenerativeAI } from '@google/generative-ai';

const API_KEY = 'AIzaSyCfDLK5EvsJaIkBf-CcTRA05QnZVa-K54o';

async function testGemini() {
  try {
    console.log('Testing Gemini API connection...');
    const genAI = new GoogleGenerativeAI(API_KEY);
    
    // Test with gemini-2.0-flash-exp
    console.log('\n1. Testing gemini-2.0-flash-exp...');
    try {
      const model = genAI.getGenerativeModel({ model: 'gemini-2.0-flash-exp' });
      const result = await model.generateContent('Say hello');
      console.log('✅ gemini-2.0-flash-exp works!');
      console.log('Response:', result.response.text());
    } catch (err) {
      console.log('❌ gemini-2.0-flash-exp failed:', err.message);
    }

    // Test with gemini-1.5-flash
    console.log('\n2. Testing gemini-1.5-flash...');
    try {
      const model = genAI.getGenerativeModel({ model: 'gemini-1.5-flash' });
      const result = await model.generateContent('Say hello');
      console.log('✅ gemini-1.5-flash works!');
      console.log('Response:', result.response.text());
    } catch (err) {
      console.log('❌ gemini-1.5-flash failed:', err.message);
    }

    // Test with gemini-pro
    console.log('\n3. Testing gemini-pro...');
    try {
      const model = genAI.getGenerativeModel({ model: 'gemini-pro' });
      const result = await model.generateContent('Say hello');
      console.log('✅ gemini-pro works!');
      console.log('Response:', result.response.text());
    } catch (err) {
      console.log('❌ gemini-pro failed:', err.message);
    }

  } catch (error) {
    console.error('❌ API Key Error:', error.message);
  }
}

testGemini();
